app/main.py

```diff
@@ -1,2 +1,54 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.load_factor = 2 / 3
+        self.threshold = int(self.capacity * self.load_factor)
+        self.table = [[] for _ in range(self.capacity)]
+
+    def table_resize(self) -> None:
+        hash_table = self.table.copy()
+        self.capacity *= 2
+        self.size = 0
+        self.threshold = int(self.capacity * self.load_factor)
+        self.table = [[] for _ in range(self.capacity)]
+
+        for node in hash_table:
+            if node:
+                self[node[0]] = node[2]
+
+    def __getitem__(self, key: Hashable) -> Any:
+        hashed_key = hash(key)
+        index = hashed_key % self.capacity
+        nodes = self.table
+
+        while nodes[index]:
+            if nodes[index][0] == key and nodes[index][1] == hashed_key:
+                return nodes[index][2]
+            index = (index + 1) % self.capacity
+        raise KeyError
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size == self.threshold:
+            self.table_resize()
+
+        hashed_key = hash(key)
+        index = hashed_key % self.capacity
+        nodes = self.table
```

Comment:
Why do you save `self.table` into `nodes`?

Reply:
For me it's easier to perceive a code(logicalwise), and line places in range 79 characters